src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTrigger.java

```diff
@@ -292,17 +289,24 @@ private boolean hasCauseFromTheSamePullRequest(@Nullable List<Cause> causes, @Nu
 
     @Override
     public void run() {
-        if(this.getBuilder().getProject().isDisabled()) {
-            logger.info(format("Build Skip (%s).", getBuilder().getProject().getName()));
-        } else {
-            logger.info(format("Build started (%s).", getBuilder().getProject().getName()));
-            this.stashPullRequestsBuilder.run();
+        if (stashPullRequestsBuilder == null) {
+            logger.info("Not ready to run.");
+            return;
         }
-        this.getDescriptor().save();
+
+        AbstractProject project = stashPullRequestsBuilder.getProject();
+        if (project.isDisabled()) {
+            logger.fine(format("Project disabled, skipping build (%s).", project.getName()));
+            return;
+        }
+
+        stashPullRequestsBuilder.run();
+        getDescriptor().save();
```

Comment:
It might be a noob question, but I don't understand why we need to save the descriptor at this point?

Reply:
I have not idea, it was there and I didn't want to change it.

Reply:
OK